### PR TITLE
gh-2130 Fix SCDF shell authentication does not work when accessing SCDF behind the proxy

### PR DESCRIPTION
**Pull request by ghillert**
* Add tests

resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/2130

### CODE REVIEW

File: spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/common/ConfigCommands.java [hunk 1]
```diff
@@ -216,8 +216,7 @@ public String target(
 		try {
 			this.targetHolder.setTarget(new Target(targetUriString, targetUsername, targetPassword, skipSslValidation));
 
-			final HttpClientConfigurer httpClientConfigurer = HttpClientConfigurer.create()
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/common/ConfigCommands.java]: I think we have an issue with the `proxy-username` option configuration here. 
```
 			@CliOption(mandatory = false, key = {
 					"proxy-username" }, help = "the username for authenticated access to the secured proxy server (valid only with a "
 					+ "username)", specifiedDefaultValue = Target.DEFAULT_SPECIFIED_PASSWORD, unspecifiedDefaultValue = Target.DEFAULT_UNSPECIFIED_PASSWORD) String proxyUsername,
```

Comment (ghillert) [reply spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/common/ConfigCommands.java]: Made the help wording better

**Comment by ghillert** [comment]
Rebased with master + addressed comment

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Merged as d0dba6f1

**Pull request merged**